python/patch-airsense.py

```diff
@@ -0,0 +1,490 @@
+
+        #Y size
+        self.fw[(structaddr + 2):(structaddr + 4)] = list(struct.pack('H', image.meta_ysize))
+
+        #'BytesPerLine' size
+        self.fw[(structaddr + 4):(structaddr + 6)] = list(struct.pack('H', image.meta_bytesper))
+
+        
+        # We leave bitsperpixel alone - should be '0'
+        #self.fw[structaddr + 6]
+        #self.fw[structaddr + 7]
+        
+        #Pointer to pixels
+        self.fw[(structaddr + 8):(structaddr + 12)] = list(struct.pack('I', pixeladdr + 0x08000000))
+        
+        #Pointer to pallete
+        self.fw[(structaddr + 12):(structaddr + 16)] = list(struct.pack('I', palletaddr + 0x08000000))
+        
+        #Pointer to function for drawing/decoding (not changed)
+        #self.fw[(structaddr + 16):(structaddr + 24)]
+        
+        #Copy pixel data over as well
+        self.patch(image.pixels, pixeladdr)
+        
+        #Pallete needs a little support struct to feel better
+        self.fw[(palletaddr + 0):(palletaddr + 4)] = list(struct.pack('I', image.pallete_numberentries))
+        self.fw[(palletaddr + 4):(palletaddr + 8)] = list(struct.pack('I', image.pallete_numbertransp))
+        self.fw[(palletaddr + 8):(palletaddr + 12)] = list(struct.pack('I', palletaddr + 16 + 0x08000000))
+
+        #Copy pallete over where we expect it
+        for i in range(0, len(image.pallete)):
+            self.fw[(palletaddr + 16 + (i*4)):(palletaddr + 16 + (i*4 + 4))] = list(struct.pack('I', image.pallete[i]))
+
+    def write_output(self, filename, overwrite=False):
+        if os.path.exists(filename) and (overwrite == False):
+            raise IOError("File " + filename + "exists already.")
+    
+        f = open(filename, "wb")
+        f.write(bytes(self.fw))
+        f.close()
+        
+    def prepare_bin(self, filename):
+        """Uses .lst file to find symbols - could use ELF too put requires additional dependancy"""
+        
+        f = open(filename + ".lst", "rb")
+        lst = f.read()
+        f.close()
+        
+        f = open(filename + ".bin", "rb")
+        bin = f.read()
+        f.close()
+        
+        #Find 'start' symbol we assume each file uses
+        addr_offset = re.search(b'\.text:[0-F]{8} start', lst, re.IGNORECASE).group(0)
+        
+        #addr should look like this now - .text:00000000 start        
+        addr_offset = addr_offset.split(b':')[1].split(b' ')[0]
+        addr_offset = int(addr_offset, 16)
+        
+        return addr_offset, bin
+
+class ASUnits(object):
+    def __init__(self, name, catalog_no, sw_ver, hash):
+        self.name = name
+        self.catalog_no = catalog_no
+        self.sw_ver = sw_ver
+        self.hash = hash
+
+class ASFirmwarePatches(object):
+    """This class contains the actual patching scripts for specific items"""
+
+    known_units = [
+        ASUnits("AirSense 10 Autoset", "37028", "SX567-0401", "533b91127aa22e05b933db203ad56c449dc12a8c3fd62f57bd88c472a8061775"),
```

Comment:
Can you add SX567-0306?

Reply:
I may as well put in a request for "AirCurve 10 VAuto",  "37051", "SX567-0401", "6790b548e0b37c57bc118772d4a04a599c0b74b16cd92e821071b9c7ba5ab711"